### PR TITLE
Improve role

### PR DESCRIPTION
**Pull request by jordiclariana**

### CODE REVIEW

File: tasks/login.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+- name: Set Vault root token fact
+  set_fact:
+    vault_root_token: "{{ vault_new_root_token }}"
+  when: vault_new_root_token is defined and vault_root_token is not defined
+
+- block:
+    - name: Read vault token from token file
+      slurp:
+        src: /root/.vault-token
```
Comment (tutte) [inline tasks/login.yml:29]: This should be defined in a var

File: templates/config.hcl.j2 [hunk 2]
```diff
@@ -1,5 +1,16 @@
 pid_file = "/var/run/vault/vault.pid"
 
+{% if vault_web_ui %}
+ui = true
+{% endif %}
+
+{% if vault_api_addr is defined %}
```
Comment (tutte) [inline templates/config.hcl.j2:7]: Is better to check if this is not False
```suggestion
{% if vault_api_addr %}
```

File: templates/config.hcl.j2 [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+api_addr = "{{ vault_public_api_addr }}"
+{% endif %}
+{% if vault_cluster_addr is defined %}
```
Comment (tutte) [inline templates/config.hcl.j2:10]: Same as above

**Review by tutte — CHANGES_REQUESTED** [review]

File: tasks/login.yml (hunk 1 above)
Comment (jordiclariana) [reply tasks/login.yml:29]: This file is created by `vault` client, it won't change neither we can decide where is goes, this is why I hardcoded it. I wouldn't change it

File: templates/config.hcl.j2 (hunk 2 above)
Comment (jordiclariana) [reply templates/config.hcl.j2:7]: Unfortunately that is dangerous. Having this playbook:

```
- hosts: all
  connection: local
  vars:
    test_ok: ""
    test_not_ok: "blah"

  tasks:
    - debug: msg="test_ok exists and empty"
      when: test_ok
      ignore_errors: yes
    - debug: msg="test_ok exists"
      when: test_ok is defined
      ignore_errors: yes
    - debug: msg="test_not_ok exists and empty"
      when: test_not_ok
      ignore_errors: yes
    - debug: msg="test_not_ok exists"
      when: test_not_ok is defined
      ignore_errors: yes
```

This is the output:

```

PLAY [all] ***********************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************
ok: [localhost]

TASK [debug] *********************************************************************************************************************************************
skipping: [localhost]

TASK [debug] *********************************************************************************************************************************************
ok: [localhost] => {
    "msg": "test_ok exists"
}

TASK [debug] *********************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'test_not_ok' failed. The error was: error while evaluating conditional (test_not_ok): 'blah' is undefined\n\nThe error appears to have been in '/home/jordi/Repos/stuart/infrastructure/ansible/a.test': line 14, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n      ignore_errors: yes\n    - debug: msg=\"test_not_ok exists and empty\"\n      ^ here\n"}
...ignoring

TASK [debug] *********************************************************************************************************************************************
ok: [localhost] => {
    "msg": "test_not_ok exists"
}

PLAY RECAP ***********************************************************************************************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0   
```

When a variable is empty string condition evaluates to false.
When a variable is empty, filter is defined evaluates to true.
When a variable is not empty string, it throws an error.
When a variable is not empty string, filter is defined evaluates to true.

The 3rd conditions is the problematic one, hence I wouldn't advice this change.

**Review by vide — APPROVED** [review]
LGTM

File: templates/config.hcl.j2 (hunk 2 above)
Comment (tutte) [reply templates/config.hcl.j2:7]: I see and understand your point. Keep in mind this can't happen if the var is defined in the defaults.yml

**Pull request merged**